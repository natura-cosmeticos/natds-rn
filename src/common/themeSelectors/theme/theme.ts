import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
