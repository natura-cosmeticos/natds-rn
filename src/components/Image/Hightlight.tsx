import React from 'react'
import { View, StyleSheet } from 'react-native'
import {
  Defs, Rect, LinearGradient, Stop, Svg
} from 'react-native-svg'
import { withTheme } from 'styled-components/native'
import { Theme } from '@naturacosmeticos/natds-themes/react-native'
import { ImageBaseProps } from './Image.types'

type FadeProps = {
  theme: Theme;
} & Pick<ImageBaseProps, 'fade' | 'variant' | 'source'>

export const getFadePosition = (fade?) => {
  const positions = {
    left: {
      x1: '1', y1: '0', x2: '0', y2: '0'
    },
    top: {
      x1: '0', y1: '1', x2: '0', y2: '0'
    },
    right: {
      x1: '0', y1: '0', x2: '1', y2: '0'
    },
    bottom: {
      x1: '0', y1: '0', x2: '0', y2: '1'
    }
  }

  return positions[fade]
}

export const isHighlight = (props: ImageBaseProps): props is ImageBaseProps => props.variant === 'highlight'
const defaultPosition = {
  x1: '0', y1: '0', x2: '0', y2: '0'
}

export const Fade = (props: FadeProps) => {
  const { theme, fade } = props

  const fadePosition = isHighlight(props) && fade === undefined
    ? defaultPosition : getFadePosition(fade)
  const opacity = isHighlight(props) && fade === undefined
    ? theme.opacity.medium : theme.opacity.veryHigh

  return (
    <View style={{
      top: 0, position: 'absolute', width: '100%', height: '100%', zIndex: 1
    }}
    >
      <Svg height="100%" width="100%" style={StyleSheet.absoluteFillObject}>
        <Defs>
          <LinearGradient id="fade_overlay" {...fadePosition}>
            <Stop offset="1" stopColor={theme.color.highlight} stopOpacity={opacity} />
            <Stop offset="0" stopColor={theme.color.highlight} stopOpacity="0" />
          </LinearGradient>
        </Defs>
        <Rect width="100%" height="100%" fill="url(#fade_overlay)" />
      </Svg>
    </View>
  )
}

export const Highlight = withTheme(Fade)
