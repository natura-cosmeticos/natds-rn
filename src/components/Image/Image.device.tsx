import { storiesOf } from '@storybook/react-native'
import {
  Default, Variants, Radius, Fade, Content, Interactive
} from './Image.stories'

storiesOf('Image', module)
  .add('Default', Default)
  .add('Variants', Variants)
  .add('Radius', Radius)
  .add('Fade', Fade)
  .add('Content', Content)
  .add('Interactive', Interactive)
