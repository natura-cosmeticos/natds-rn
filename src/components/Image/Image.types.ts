import {
  ImageProps as ImageNativeProps, ImagePropsAndroid, ImagePropsIOS, ImageSourcePropType
} from 'react-native'

type IOSPlatformProps = Pick<ImagePropsIOS,
  | 'capInsets'
  | 'onPartialLoad'
>

type AndroidPlatformProps = Pick<ImagePropsAndroid,
  | 'fadeDuration'
>

type MultiPlatformProps = Pick<ImageNativeProps,
  | 'accessible'
  | 'accessibilityLabel'
  | 'blurRadius'
  | 'defaultSource'
  | 'loadingIndicatorSource'
  | 'onError'
  | 'onLayout'
  | 'onLoad'
  | 'onLoadEnd'
  | 'onLoadStart'
  | 'onProgress'
  | 'resizeMode'
  | 'style'
>

export interface ImageBaseProps extends
  MultiPlatformProps,
  IOSPlatformProps,
  AndroidPlatformProps {
  /**
   * Optional ID for testing.
   */
  testID?: string;

  /**
   * Optional attribute will bring the component arranged differently for each type.
   * @default 'standard'
   */
  variant?: 'standard' | 'highlight';

  /**
   * Optional property defines the radius of the element's corners.
   * @default 'none'
   */
  radius?: 'none' | 'medium' | 'circle';

  /**
   * Optional attribute to define the side of the fade effect.
   * @default 'bottom'
   */
  fade?: 'top' | 'left' | 'right' | 'bottom';

  /**
   * Can receive any component as a child to be used as content
   */
  children?: React.ReactNode;

  /**
   * The image source either a remote URL or a local file resource.
   */
  source: string | ImageSourcePropType;
}

export type ImageProps = ImageBaseProps
