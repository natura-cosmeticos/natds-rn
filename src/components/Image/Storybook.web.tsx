import React from 'react'

export const Svg = (props) => {
  const { xml } = props

  return (
    // eslint-disable-next-line react/no-danger
    <div dangerouslySetInnerHTML={{ __html: xml }} />
  )
}

export const Rect = (props) => {
  const { xml } = props

  return (
    // eslint-disable-next-line react/no-danger
    <div dangerouslySetInnerHTML={{ __html: xml }} />
  )
}

export const Defs = (props) => {
  const { xml } = props

  return (
    // eslint-disable-next-line react/no-danger
    <div dangerouslySetInnerHTML={{ __html: xml }} />
  )
}

export const LinearGradient = (props) => {
  const { xml } = props

  return (
    // eslint-disable-next-line react/no-danger
    <div dangerouslySetInnerHTML={{ __html: xml }} />
  )
}

export const Stop = (props) => {
  const { xml } = props

  return (
    // eslint-disable-next-line react/no-danger
    <div dangerouslySetInnerHTML={{ __html: xml }} />
  )
}
