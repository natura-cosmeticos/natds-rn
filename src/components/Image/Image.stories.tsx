import React from 'react'
import { View, Text, Platform } from 'react-native'
import { select, text } from '@storybook/addon-knobs'
import { StoryContainer, StoryWrapper } from '../../common/HelperComponents/StoryContainer'

import { Image } from './Image'
import NatAvatar from '../../assets/images/nat_avatar.jpg'
import { Badge } from '../Badge'
import { Tag } from '../Tag'

const webStoryMessage = Platform.OS === 'web' ? ' (the effect can only be rendered within the native platforms)' : ''

const description = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ Standard
  - ✅ Highlight

With the following attribute statuses:

  - **Attributes:**
  - **Radius:**
    - ✅ \`None\`
    - ✅ \`Medium\`
    - ✅ \`Circle\`
  - **Fade:**
    - ✅ \`Left\`
    - ✅ \`Top\`
    - ✅ \`Right\`
    - ✅ \`Bottom\`

  - ✅ **Fallback** (is available using the native onError method)

- - -
`

export default {
  component: Image,
  parameters: {
    componentSubtitle: 'Informs the user of image content with possible resources for greater interaction',
    docs: {
      extractComponentDescription: description
    }
  },
  title: 'Components|Image'
}

export const Default = () => (
  <StoryWrapper title="Default">
    <View style={{ width: 100, height: 100 }}>
      <Image source={NatAvatar} />
    </View>
  </StoryWrapper>
)

export const Variants = () => (
  <StoryWrapper title="Variants">
    <StoryContainer title="Standard">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} />
      </View>
    </StoryContainer>
    <StoryContainer title={`Highlight${webStoryMessage}`}>
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} variant="highlight" />
      </View>
    </StoryContainer>
  </StoryWrapper>
)

export const Radius = () => (
  <StoryWrapper title="Radius">
    <StoryContainer title="None">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} />
      </View>
    </StoryContainer>
    <StoryContainer title="Medium">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} radius="medium" />
      </View>
    </StoryContainer>
    <StoryContainer title="Circle">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} radius="circle" />
      </View>
    </StoryContainer>
  </StoryWrapper>
)

export const Fade = () => (
  <StoryWrapper title={`Fade${webStoryMessage}`}>
    <StoryContainer title="Left">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} variant="highlight" fade="left" />
      </View>
    </StoryContainer>
    <StoryContainer title="Top">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} variant="highlight" fade="top" />
      </View>
    </StoryContainer>
    <StoryContainer title="Right">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} variant="highlight" fade="right" />
      </View>
    </StoryContainer>
    <StoryContainer title="Bottom">
      <View style={{ width: 100, height: 100 }}>
        <Image source={NatAvatar} variant="highlight" fade="bottom" />
      </View>
    </StoryContainer>
  </StoryWrapper>
)

export const Content = () => (
  <StoryWrapper title="Content">
    <StoryContainer title="Components">
      <View style={{ width: 200, height: 200 }}>
        <Image source={NatAvatar} variant="highlight" fade="top">
          <View style={{ flexDirection: 'row', alignContent: 'center', justifyContent: 'center' }}>
            <View style={{ alignContent: 'center', justifyContent: 'center', marginRight: 8 }}>
              <Badge variant="pulse" />
            </View>
            <Tag text="Natura Design System" borderPosition="right" color="alert" />
          </View>
        </Image>
      </View>
    </StoryContainer>
    <StoryContainer title="Text">
      <View style={{ width: 200, height: 200 }}>
        <Image source={NatAvatar} variant="highlight" fade="top">
          <Text style={{ color: '#fff' }}>Text, Icons, Components...</Text>
        </Image>
      </View>
    </StoryContainer>
  </StoryWrapper>
)

export const Interactive = () => (
  <StoryWrapper title="Interactive">
    <View style={{ width: 300, height: 300 }}>
      <Image
        source={text('Source', NatAvatar)}
        variant={select('Variant', ['standard', 'highlight'], 'standard')}
        fade={select('Fade', ['left', 'top', 'right', 'bottom'], 'bottom')}
        radius={select('Radius', ['none', 'medium', 'circle'], 'none')}
      >
        <Text style={{ color: '#fff' }}>{text('Children', 'Text, Icons, Components...')}</Text>
      </Image>
    </View>
  </StoryWrapper>
)
