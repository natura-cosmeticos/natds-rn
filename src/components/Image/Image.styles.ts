import styled from 'styled-components/native'
import { Dimensions } from 'react-native'
import { Theme } from '@naturacosmeticos/natds-themes/react-native'
import { ImageBaseProps } from './Image.types'

type WrapperStyleProps = {
  theme: Theme;
} & Pick<ImageBaseProps, 'radius'>

export const getBorderRadius = ({ theme, radius = 'none' }: WrapperStyleProps) => {
  const borders = {
    none: 0,
    medium: theme.borderRadius.medium,
    circle: Dimensions.get('window').width * 2
  }

  return borders[radius] || borders.none
}

export const Wrapper = styled.View(({ theme, radius = 'none' }: WrapperStyleProps) => ({
  flex: 1,
  borderRadius: getBorderRadius({ theme, radius }),
  overflow: 'hidden'
}))

export const Image = styled.Image(() => ({
  flex: 1,
  aspectRatio: 1
}))

export const Content = styled.View(() => ({
  position: 'absolute',
  top: 0,
  zIndex: 2,
  marginLeft: 8,
  marginRight: 8,
  marginTop: 8,
  alignItems: 'center',
  justifyContent: 'center'
}))
