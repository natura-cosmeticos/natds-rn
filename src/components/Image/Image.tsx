import React, { FunctionComponent, forwardRef } from 'react'

import { ImageProps as ImageNativeProps } from 'react-native'
import * as S from './Image.styles'
import { ImageProps } from './Image.types'
import { Highlight } from './Hightlight'

export const Image:
  FunctionComponent<ImageProps> = forwardRef<ImageNativeProps, ImageProps>((props, ref) => {
    const {
      variant = 'standard', radius = 'none', fade, children, testID, ...rest
    } = props

    return (
      <S.Wrapper radius={radius} testID={testID}>
        {radius !== 'circle' && (
        <S.Content>
          {children}
        </S.Content>
        )}
        {variant === 'highlight' && <Highlight fade={fade} variant={variant} />}
        <S.Image ref={ref} {...rest} />
      </S.Wrapper>
    )
  })
