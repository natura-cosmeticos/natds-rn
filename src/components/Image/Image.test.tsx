import React from 'react'

import { Text } from 'react-native'
import { renderWithTheme } from '../../../test/testHelpers'
import { Image } from './Image'
import NatAvatar from '../../assets/images/nat_avatar.jpg'
import { getFadePosition, Highlight } from './Hightlight'
import { getBorderRadius } from './Image.styles'
import theme from '../../common/themeSelectors/theme/mock-theme.json'

describe('Image Component', () => {
  it('should render with default variant', () => {
    const { toJSON } = renderWithTheme(<Image source={NatAvatar} testID="default" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with highlight variant', () => {
    const { toJSON } = renderWithTheme(<Image source={NatAvatar} variant="highlight" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with bottom fade', () => {
    const { toJSON } = renderWithTheme(<Image source={NatAvatar} variant="highlight" fade="bottom" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with a text children', () => {
    const { toJSON } = renderWithTheme(
      <Image source={NatAvatar}>
        <Text>Hello</Text>
      </Image>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the image component with radius different to default', () => {
    const { getByTestId } = renderWithTheme(<Image source={NatAvatar} radius="medium" testID="medium-radius" />)

    const result = getByTestId('medium-radius')

    expect(result?.props).toHaveProperty('radius', 'medium')
    expect(result).toHaveStyle({ borderRadius: 4 })
  })

  it('should render a highlight effect class', () => {
    const { toJSON } = renderWithTheme(<Highlight variant="highlight" fade="bottom" />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should return 0 value when not receive a radius props', () => {
    const result = getBorderRadius({ theme })

    expect(result).toEqual(0)
  })

  it('should return a object value when receive a bottom fade props', () => {
    const result = getFadePosition('bottom')

    expect(result).toMatchObject({
      x1: '0', y1: '0', x2: '0', y2: '1'
    })
  })

  it('should return a undefined when not receive fade props', () => {
    const result = getFadePosition()

    expect(result).toEqual(undefined)
  })
})
