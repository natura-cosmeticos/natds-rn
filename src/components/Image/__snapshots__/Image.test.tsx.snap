// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render a highlight effect class 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <RNSVGSvgView
    bbHeight="100%"
    bbWidth="100%"
    focusable={false}
    height="100%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              0,
              1,
              -872415232,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="fade_overlay"
          x1="0"
          x2="0"
          y1="0"
          y2="1"
        />
      </RNSVGDefs>
      <RNSVGRect
        fill={
          Array [
            1,
            "fade_overlay",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        width="100%"
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Image Component should render with a text children 1`] = `
<View
  radius="none"
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "position": "absolute",
          "top": 0,
          "zIndex": 2,
        },
      ]
    }
  >
    <Text>
      Hello
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/nat_avatar.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`Image Component should render with bottom fade 1`] = `
<View
  radius="none"
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "position": "absolute",
          "top": 0,
          "zIndex": 2,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                0,
                1,
                -872415232,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="fade_overlay"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "fade_overlay",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/nat_avatar.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`Image Component should render with default variant 1`] = `
<View
  radius="none"
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "hidden",
      },
    ]
  }
  testID="default"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "position": "absolute",
          "top": 0,
          "zIndex": 2,
        },
      ]
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/nat_avatar.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`Image Component should render with highlight variant 1`] = `
<View
  radius="none"
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "position": "absolute",
          "top": 0,
          "zIndex": 2,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                0,
                1,
                2046820352,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="fade_overlay"
            x1="0"
            x2="0"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "fade_overlay",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/nat_avatar.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
